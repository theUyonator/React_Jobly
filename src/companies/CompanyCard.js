import React from "react";
import { Link } from "react-router-dom";

import "./CompanyCard.css";

/**This company shows information about a
 * company and is rendered by the 
 * companylist component.
 */

 function CompanyCard ({name, description, logoUrl, handle}){
    return (
        <Link className="CompanyCard card" to={`/companies/${handle}`}>
            <div className="card-body">
                <h6 className="card-title">
                    {name}
                    {logoUrl && <img src={logoUrl}
                                     alt={name}
                                     className="float-right ml-5"
                                />
                    }

                </h6>
                <p><small>{description}</small></p>
            </div>
        </Link>
    );
 }

 export default CompanyCard;