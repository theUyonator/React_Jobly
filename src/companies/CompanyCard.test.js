import React from "react";
import { render } from "@testing-library/react";
import CompanyCard from "./CompanyCard";
import { MemoryRouter } from "react-router";

// smoke test
test("renders without crashing", function() {
    render(<CompanyCard />);
  });

test("matches snapshot with company logo", function () {
    const { asFragment } = render(
        <MemoryRouter>
            <CompanyCard
                handle="abc"
                name="ABC Apparel"
                description="ABC Apparel is an exceptional outfit"
                logo_url="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcSPpsiqd2mYQzgOEhTWnQ3OktQZjZh3__OJrw&usqp=CAU"
            />
        </MemoryRouter>,
    );
    expect(asFragment()).toMatchSnapshot();
});


test("matches snapshot without company logo", function () {
    const { asFragment } = render(
        <MemoryRouter>
            <CompanyCard
                handle="abc"
                name="ABC Apparel"
                description="ABC Apparel is an exceptional outfit"
            />
        </MemoryRouter>,
    );
    expect(asFragment()).toMatchSnapshot();
});