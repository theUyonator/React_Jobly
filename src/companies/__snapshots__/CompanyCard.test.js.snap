// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot with company logo 1`] = `
<DocumentFragment>
  <a
    class="CompanyCard card"
    href="/companies/abc"
  >
    <div
      class="card-body"
    >
      <h6
        class="card-title"
      >
        ABC Apparel
      </h6>
      <p>
        <small>
          ABC Apparel is an exceptional outfit
        </small>
      </p>
    </div>
  </a>
</DocumentFragment>
`;

exports[`matches snapshot without company logo 1`] = `
<DocumentFragment>
  <a
    class="CompanyCard card"
    href="/companies/abc"
  >
    <div
      class="card-body"
    >
      <h6
        class="card-title"
      >
        ABC Apparel
      </h6>
      <p>
        <small>
          ABC Apparel is an exceptional outfit
        </small>
      </p>
    </div>
  </a>
</DocumentFragment>
`;
